back/engines/commercial/insights/app/policies/insights/view_policy.rb

```diff
@@ -15,24 +15,44 @@ def resolve
         return scope.all if user.admin?
         raise Pundit::NotAuthorizedError unless user.project_moderator?
 
-        scope.joins(:data_sources).where(data_sources: { origin_id: user.moderatable_project_ids })
+        scope.where.not(
+          # views with data sources that are not moderated by the user
+          id: scope.joins(:data_sources).where.not(data_sources: { origin_id: user.moderatable_project_ids })
+        )
       end
     end
 
     def show?
-      user&.active_admin_or_moderator?(record.scope.id)
+      allowed?
     end
 
     def create?
-      user&.active_admin_or_moderator?(record.scope.id)
+      allowed?
     end
 
     def update?
-      user&.active_admin_or_moderator?(record.scope.id)
+      allowed?
     end
 
     def destroy?
-      user&.active_admin_or_moderator?(record.scope.id)
+      allowed?
+    end
+
+    private
+
+    def allowed?
+      return unless active?
```

Comment:
The early `return` here would be `nil`, right?

I understand that this evaluates to `false`, but I'd expect a method ending with a question mark (and being a policy) to either return `true` or `false` explicitly here.

```suggestion
      return false unless active?
```

Reply:
Committing this one, but in general, I really have no strong feelings about returning something that's not a boolean from a predicate method. To me, the question mark just there to say that you can use the result as an operand in a boolean expression. But I defo get your point.

Reply:
Fair point. :+1: